lms/djangoapps/teams/serializers.py

```diff
@@ -108,8 +112,43 @@ class Meta(object):
         read_only_fields = ("date_joined",)
 
 
-class TopicSerializer(serializers.Serializer):
-    """Serializes a topic."""
+class BaseTopicSerializer(serializers.Serializer):
+    """Serializes a topic without team_count."""
     description = serializers.CharField()
     name = serializers.CharField()
     id = serializers.CharField()  # pylint: disable=invalid-name
+
+
+class TopicSerializer(BaseTopicSerializer):
+    """
+    Adds team_count to the basic topic serializer.  Use only when
+    serializing a single topic.  When serializing many topics, use
+    `PaginatedTopicSerializer` to avoid O(N) SQL queries.
+    """
+    team_count = serializers.SerializerMethodField('get_team_count')
+
+    def get_team_count(self, topic):
+        """Get the number of teams associated with this topic"""
+        return CourseTeam.objects.filter(topic_id=topic['id']).count()
+
+
+class PaginatedTopicSerializer(PaginationSerializer):
```

Comment:
I feel like that test only covers the basic path. I'd like to see a test for what happens if there are no topics, one for if there are topics that have no counts, and also what happens if there are counts for a topic that isn't being returned. It would also be good to verify that the pagination logic works, i.e. does this still behave correctly if this is only a subset of all the topics.